app/controllers/api_users_controller.rb

```diff
@@ -64,7 +63,8 @@ def api_user_params
     ).sanitise
   end
 
-  def permitted_user_params(params)
-    params.permit(:email, :name, supported_permission_ids: [])
+  def permitted_user_params(filtered_params)
```

Comment:
If I understand correctly then the `create` action only allows `email` and `name` params, and the `update` action only allows `supported_permission_ids` param. If that is correct then it might be clearer to either specify those allowed  params in the actions, or split this into something like `permitted_user_params_for_create` and `permitted_user_params_for_update`.

Reply:
Good idea!

Reply:
I've remembered more about why I did this. The actions are actually using another method, `#api_user_params` which in turn calls this method, `#permitted_user_params`, so it's quite awkward to do what you're suggesting. I'll give it one more go...

Reply:
I think I've cracked it and in doing so spotted a minor security issue!

* https://github.com/alphagov/signon/pull/2593/commits/e74494c5aa1d48834a5c7d8b2612791571430d76
* https://github.com/alphagov/signon/pull/2593/commits/e9b3acbfc0ce31bea6f7b1151ca90f7d57c4f369
* https://github.com/alphagov/signon/pull/2593/commits/58d0433fceadefb114eae14996caf745e004f809
* https://github.com/alphagov/signon/pull/2593/commits/33bffd1fbc03143f78616da8cf1ce31d49788ade